modules/magnetometer/src/main/java/com/gluonhq/attach/magnetometer/impl/IOSMagnetometerService.java

```diff
@@ -45,6 +46,10 @@
     private static final ReadOnlyObjectWrapper<MagnetometerReading> reading = new ReadOnlyObjectWrapper<>();
 
     public IOSMagnetometerService() {
+        Utils.runOnAppThread(this::setupObserver);
```

Comment:
These and other services simply should fail when invoked before the JavaFX framework is launched.